src/platform/Darwin/WiFi/NetworkCommissioningWiFiDriver.mm

```diff
@@ -72,18 +73,23 @@ bool Next(WiFiScanResponse & scanResponse) override
     void CopyNetworkInformationTo(WiFiScanResponse & destination, CWNetwork * source)
     {
         destination.security = GetWiFiSecurity(source);
-        destination.ssidLen = [source.ssid length];
-        destination.channel = source.wlanChannel.channelNumber;
+        destination.channel = static_cast<uint16_t>(source.wlanChannel.channelNumber);
         destination.wiFiBand = GetWiFiBand(source.wlanChannel);
-        destination.rssi = source.rssiValue;
-        CopyStringTo(destination.ssid, source.ssid, DeviceLayer::Internal::kMaxWiFiSSIDLength);
-        CopyStringTo(destination.bssid, source.bssid, DeviceLayer::Internal::kWiFiBSSIDLength);
-    }
-
-    void CopyStringTo(uint8_t * destination, NSString * source, size_t maxLength)
-    {
-        NSData * data = [source dataUsingEncoding:NSUTF8StringEncoding];
-        memcpy(destination, [data bytes], std::min([data length], maxLength));
+        destination.rssi = static_cast<int8_t>(source.rssiValue);
+
+        NSData * ssidData = source.ssidData;
+        destination.ssidLen = static_cast<uint8_t>(std::min(ssidData.length, DeviceLayer::Internal::kMaxWiFiSSIDLength));
+        memcpy(destination.ssid, ssidData.bytes, destination.ssidLen);
+
+        // source.bssid looks like "00:00:00:00:00:00" if it's not nil.
+        NSString * bssid = source.bssid;
+        // 3 chars per byte, except the last byte.
+        if (bssid.length == 3 * sizeof(destination.bssid) - 1) {
```

Comment:
Should an error be logged if the bssid isn't nil, but also ins't the right length?

Reply:
Probably a good idea...  In practice, it's always going to be nil, I suspect.

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/25361 to do that.